tests/http_integration.rs

```diff
@@ -540,6 +541,35 @@ mod http_integration {
 		Ok(())
 	}
 
+	#[test(tokio::test)]
+	async fn sql_endpoint_with_compression() -> Result<(), Box<dyn std::error::Error>> {
+		let (addr, _server) = common::start_server_with_defaults().await.unwrap();
+		let url = &format!("http://{addr}/sql");
+
+		// Prepare HTTP client
+		let mut headers = reqwest::header::HeaderMap::new();
+		headers.insert("NS", "N".parse()?);
+		headers.insert("DB", "D".parse()?);
+		headers.insert(header::ACCEPT, "application/json".parse()?);
+		headers.insert(header::ACCEPT_ENCODING, "gzip".parse()?);
+
+		let client = reqwest::Client::builder()
+			.connect_timeout(Duration::from_millis(10))
+			.gzip(false) // So that the content-encoding header is not removed by Reqwest
+			.default_headers(headers.clone())
+			.build()?;
+
+		// Check that the content is gzip encoded
+		{
```

Comment:
I dont think it needs to be in a separate block? Happy if it stays or goes, but my preference is no blocks unless its explicit scope.

Reply:
I assume this was a copy&paste from other test cases. I followed this approach in other tests to avoid the `res` value be leaked into asserts by mistake.

Imagine running several requests, all of them storing the result in `res` and then doing asserts. If the `res` is not properly reset, or you refactor later on, there's the risk of the value being used in asserts not intended for it.

Might be a bit of an edge case, but is a handy way to have "isolated" test cases inside test cases. Open to alternatives though!

Reply:
I totally agree with you @phughk in the production code.
For tests, I like using it to improve the readability too, as it has no impact.

Reply:
Ah all good, agree with `res` use case :) Resolving